src/intervals/interval_operations/pointwise_boolean.jl

```diff
@@ -0,0 +1,204 @@
+# This file is part of the IntervalArithmetic.jl package; MIT licensed
+
+#=  This file contains functions not specified in the IEEE Std 1788-2015,
+    extending boolean operations defined for reals to intervals.
+
+    This is different from what is described in sections 9.5 of the
+    IEEE Std 1788-2015 (Boolean functions of intervals), since we deal here
+    with what is needed for drop-in replacement of floats by intervals,
+    and not what is reasonnable to do with intervals in a separate execution
+    environment.
+
+    Essentially, julia is better at composability and we are more ambitious
+    that what the standard ever expected, so we have some extra problems
+    to solve.
+
+    We use a trait system, defining the operation with an extra
+    `PointwisePolitic` argument defining how it should be handled.
+
+    The default operations use `IntervalArithmetic.pointwise_politic()`,
+    so in order to change the default behavior this function has to be
+    redefined.
+
+    We define every operations using the default `PointwisePolitic{:ternary}()`
+    and then use it to define all the others.
+=#
+"""
+    PointwisePolitic{P}
+
+Define which politic we use to extend pointwise comparison of 
+
+Valid value for the politic identifier `P` are
+    - `:is_all` : A boolean operation is extended by asking "is it true for
+        all elements of the interval(s) involved".
+        This is self-consistent, but breaks the usual rules for negation.
+        For example with this politic, `((-1..3) == 0) == false` because it
+        answers the question "are all elements in (-1..3) equal to zero".
+        However its negation is `false` too: `((-1..3) != 0) == false`.
+        This *silently* breaks any code relying on such operation in conditional
+        statements like `if x == 0 ... else ... end`.
+    - `:interval` : A pointwise boolean operation `B` return the set of all
+        possible outcome as a `BooleanInterval`.
+        This is safe but very strict, always erroring when an interval is used
+        in a conditional statement.
+    - `:ternary` (default) : With this politics return `missing` when the
+        boolean operation does not return the same answer for all elements
+        of the involved interval(s).
+        This only causes error in conditional statements when hitting `missing`.
+        When it does not error it is safe.
+
```

Comment:
I think we should have another politic identifier such that the ITF1788 tests pass, e.g. `≤(Interval(0.0, 2.0), Interval(1.0, 2.0))` or `Interval(-Inf, Inf) ≤ Interval(-Inf,Inf)` (both should return `true`). That mode (call it `:ieee1788` or `:exist_one` or something else) simply requests that one element fulfills the checked property.

Reply:
> That mode (call it :ieee1788 or :exist_one or something else) simply requests that one element fulfills the checked property.

This is not quite true... sorry. That mode is some sort of strict binary politic...

Reply:
I will have to check that in detail, because it may be it uses the symbol defined in the standard that we call `precedes` or something similar. That way we can keep the symbols for their pointwise extension.

Reply:
the standard defines `[a, b] <= [c, d]` as `a <= c && b <= d` (Table 10.3). I also think there should be a standard-compliant flavor

Reply:
Ah I see it now.

What is defined in Table 10.3 is not, I think, the extension of the real `<=` operator to interval, but some other operator that are only defined for interval, and unfortunately share the same symbol.

All function in table 10.3 should be added in `boolean.jl` (I overlooked 10.5.10), ideally under their full name (e.g. `less`) but I don't think it makes sense to have a pointwise politic for them, precisely because their are not pointwise extensions.

The standard specifically allows (somewhere) to use the different name or symbols from what it does.